test/meson.build

```diff
@@ -77,3 +86,13 @@ test(
   env : ['SHARNESS_BUILD_DIRECTORY=' + meson.build_root()],
   workdir : meson.current_source_dir(),
 )
+
+install_data(
```

Comment:
This will be installed unconditionally. Maybe guard the additional install data calls with an `if install_tests`:

``` meson
if install_tests
install_data(..)
install_subdir(..)
...
endif
```